server/src/test/java/org/opensearch/index/mapper/FlatObjectFieldTypeTests.java

```diff
@@ -0,0 +1,134 @@
+        jsonPointList.add(jsonPoint.toString());
+
+        ArrayList<String> otherJsonPointList = new ArrayList<>();
+        otherJsonPointList.add(otherJsonPoint.toString());
+
+        assertEquals(jsonPointList, fetchSourceValue(mapper, jsonPoint, null));
+        assertEquals(otherJsonPointList, fetchSourceValue(mapper, otherJsonPoint, null));
+
+    }
+
+    public void testDirectSubfield() {
+        {
+            MappedFieldType flatParentFieldType = getFlatParentFieldType("field");
+
+            // when searching for "foo" in "field", the directSubfield is field._value field
+            String searchFieldName = ((FlatObjectFieldMapper.FlatObjectFieldType) flatParentFieldType).directSubfield();
+            assertEquals("field._value", searchFieldName);
+
+            MappedFieldType dynamicMappedFieldType = new FlatObjectFieldMapper.FlatObjectFieldType("bar", flatParentFieldType.name());
+            // when searching for "foo" in "field.bar", the directSubfield is field._valueAndPath field
+            String searchFieldNameDocPath = ((FlatObjectFieldMapper.FlatObjectFieldType) dynamicMappedFieldType).directSubfield();
+            assertEquals("field._valueAndPath", searchFieldNameDocPath);
+        }
+        {
+            NamedAnalyzer analyzer = new NamedAnalyzer("default", AnalyzerScope.INDEX, null);
+            MappedFieldType ft = new FlatObjectFieldMapper.FlatObjectFieldType("field", analyzer);
+            assertEquals("field._value", ((FlatObjectFieldMapper.FlatObjectFieldType) ft).directSubfield());
+        }
+    }
+
+    public void testRewriteValue() {
+        MappedFieldType flatParentFieldType = getFlatParentFieldType("field");
+
+        // when searching for "foo" in "field", the rewrite value is "foo"
+        String searchValues = ((FlatObjectFieldMapper.FlatObjectFieldType) flatParentFieldType).rewriteValue("foo");
+        assertEquals("foo", searchValues);
+
+        MappedFieldType dynamicMappedFieldType = new FlatObjectFieldMapper.FlatObjectFieldType("field.bar", flatParentFieldType.name());
+
+        // when searching for "foo" in "field.bar", the rewrite value is "field.bar=foo"
+        String searchFieldNameDocPath = ((FlatObjectFieldMapper.FlatObjectFieldType) dynamicMappedFieldType).directSubfield();
+        String searchValuesDocPath = ((FlatObjectFieldMapper.FlatObjectFieldType) dynamicMappedFieldType).rewriteValue("foo");
+        assertEquals("field.bar=foo", searchValuesDocPath);
+    }
+
+    public void testTermQuery() {
+
+        MappedFieldType flatParentFieldType = getFlatParentFieldType("field");
+
+        // when searching for "foo" in "field", the term query is directed to search "foo" in field._value field
+        String searchFieldName = ((FlatObjectFieldMapper.FlatObjectFieldType) flatParentFieldType).directSubfield();
+        String searchValues = ((FlatObjectFieldMapper.FlatObjectFieldType) flatParentFieldType).rewriteValue("foo");
```

Comment:
Aha, here what I was looking at, the `rewriteValue` says (in javadocs):

```
 /**
         * If the search key is assigned with value,
         * the dot path was used in search query, then
         * rewrite the searchValueString as the format "dotpath=value",
         * @return rewriteSearchValue
         */
```

The question I was asked earlier was when the value comes in a form of  `"dotpath=value"`? (this is just to understand this particular case).

Reply:
right, so in when there is hasdotPath/hasMappedFieldTyeNameInQueryFieldName  [here](https://github.com/opensearch-project/OpenSearch/blob/e9a25c7758a67f148ab5939d85ad25916ec0de44/server/src/test/java/org/opensearch/index/mapper/FlatObjectFieldTypeTests.java#L84C84-L84C85), it will rewrite to form of "dotpath=value"

```
 // when searching for "foo" in "field.bar", the rewrite value is "field.bar=foo"
        String searchFieldNameDocPath = ((FlatObjectFieldMapper.FlatObjectFieldType) dynamicMappedFieldType).directSubfield();
        String searchValuesDocPath = ((FlatObjectFieldMapper.FlatObjectFieldType) dynamicMappedFieldType).rewriteValue("foo");
        assertEquals("field.bar=foo", searchValuesDocPath);
```

Reply:
I will make the javadocs more clear when I resolve the conflic. Thanks @reta !!!